packages/schema-extract/README.md

```diff
@@ -0,0 +1,217 @@
+To get the linked schema of a specific file invoke `linker.flatten(fileName)`:
+
+```
+    import {createLinker} from '@ui-autotools/schema-extract';
+    const files = [
+        'myProject/src/file-a.ts',
+        'myProject/src/file-b.ts'
+    ];
+    const linker = createLinker(files);
+    const linkedSchema = linker.flatten(files[0], 'IAnimal');
+
+```
+
+## Example
+
+Take a look at the following interface:
+```
+// file-a.ts
+    export interface IAnimal {
+        name: string;
+        hasTail: boolean;
+        makeNoise: (sound: ISound) => void;
+        isTamed?: 'YES' | 'NO';
+    }
+```
+If we run the command from the usage section it will be transformed to:
+```
+{
+  "$schema": "http://json-schema.org/draft-06/schema#",
+  "$id": "/index.tsx",
+  "$ref": "common/module",
+  "properties": {},
+  "definitions": {
+    "IAnimal": {
+      "properties": {
+        "name": {
+          "type": "string"
+        },
+        "hasTail": {
+          "type": "boolean"
+        },
+        "makeNoise": {
+          "$ref": "common/function",
+          "arguments": [
+            {
+              "$ref": "#ISound",
+              "name": "sound"
+            }
+          ],
+          "returns": {
+            "$ref": "common/undefined"
+          },
+          "requiredArguments": [
+            "sound"
+          ]
+        },
+        "isTamed": {
+          "type": "string",
+          "enum": [
+            "YES",
+            "NO"
+          ]
+        }
+      },
+      "required": [
+        "name",
+        "hasTail",
+        "makeNoise"
+      ],
+      "$ref": "common/interface"
+    }
+  }
+}
+```
+
+## TS Transformer
+
+Transforms a typescript source file into JSON-Schema. 
+
+#### Usage
+
+Import the `transform` function from `@ui-autotools/schema-extract`. This function recives the following five arguments:
+
+|Name|Type|Description|
+|-------------|----|-----------|
+|checker|`typescript.TypeChecker`|A typescript checker connected to the desired files|
+|sourceFile|`typescript.SourceFile`|The typescript source of that we want to transform|
+|modulePath|`string`|The path to the module base directory|
+|projectPath|`string`|The name of the project the files are in (Will be removed in the future)|
+|pathUtil|`IFileSystemPath`|A path utility to be used to access directories and file. (You can use `path.posix`)|
+
+For example if we want to transform our own code:
+
+```
+    import path from 'path';
+    import compilerOptions from './utils';
+
+    const projectPath = '/Projects/MySecretProject/;
+    const fileName = projectPath + 'secret-stuff.ts'
+    // TS setup
+    const program = typescript.createProgram([fileName], compilerOptions);
+    const sourceFile = program.getSourceFile(fileName);
+
+    const schema = transform(program.getTypeChecker(), sourceFile, fileName, 'MySecretProject', path.posix);
+```
+
+## Linker
```

Comment:
Consider moving this part to above ts transform, but keep custom linker in the end